tests/wpt/mozilla/tests/mozilla/canvas-oversize-serialization.html

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+  <title>Serializing a large canvas does not panic</title>
+  <link rel=help href="https://html.spec.whatwg.org/multipage/#dom-canvas-todataurl">
+  <link rel=help href="https://html.spec.whatwg.org/multipage/#dom-canvas-toblob">
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+</head>
+
+<body>
+<!-- This is not a standard WPT tests, because canvas size limits are specific
+    to browsers. For us, failure to serialize depends on both canvas size limits
+    and also whether or not the image library we use (image-rs) produces an error
+    when we attempt serialization. -->
+<canvas id="canvas" width="2000000"></canvas>
```

Comment:
Is setting big enough width&height not enough to make this work in all browsers? Or do some browser simply error differently/before that?

For canvas size limit there is: https://jhildenbiddle.github.io/canvas-size/#/?id=test-results

Reply:
> Is setting big enough width&height not enough to make this work in all browsers? Or do some browser simply error differently/before that?

Since it isn't specified, I felt a bit weird about adding it to the WPT. Just to be conservative, I instead made it a Servo-specific test.